benchmarks/performancepoetry/src/main/java/com/squareup/benchmarks/performance/poetry/PerformancePoemWorkflow.kt

```diff
@@ -0,0 +1,244 @@
+import com.squareup.workflow1.ui.BuilderViewFactory
+import com.squareup.workflow1.ui.ViewFactory
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.backstack.BackStackScreen
+import com.squareup.workflow1.ui.backstack.toBackStackScreen
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+
+/**
+ * Renders a [Poem] as a [OverviewDetailScreen], whose overview is a [StanzaListRendering]
+ * for the poem, and whose detail traverses through [StanzaRendering]s.
+ */
+public class PerformancePoemWorkflow(
+  private val simulatedPerfConfig: SimulatedPerfConfig = SimulatedPerfConfig.NO_SIMULATED_PERF,
+  private val isLoading: MutableStateFlow<Boolean>
+) : PoemWorkflow, StatefulWorkflow<Poem, State, ClosePoem, OverviewDetailScreen>() {
+
+  public sealed class State {
+    // N.B. This state is a smell. We include it as possible in order to be able to config similar
+    // to real-world usage. It should be folded into [NoSelection] at the very least.
+    public object Initializing : State()
+    public data class ComplexCall(
+      val payload: Int
+    ) : State()
+
+    public data class Selected(val stanzaIndex: Int) : State()
+  }
+
+  @OptIn(WorkflowUiExperimentalApi::class)
+  public data class PoemRenderingWithLoading(
+    val screen: OverviewDetailScreen,
+    // This won't be used if we are reporting through Output.
+    val isLoading: Boolean = false
+  ) : AndroidViewRendering<PoemRenderingWithLoading> {
+    // Note that this ViewFactory does not handle the loading state at all. It just ignores it.
+    override val viewFactory: ViewFactory<PoemRenderingWithLoading>
+      get() = BuilderViewFactory(
+        PoemRenderingWithLoading::class
+      ) { initialRendering, initialViewEnvironment, contextForNewView, container ->
+        OverviewDetailContainer.buildView(
+          initialRendering = initialRendering.screen,
+          initialViewEnvironment,
+          contextForNewView,
+          container
+        )
+      }
+  }
+
+  override fun initialState(
+    props: Poem,
+    snapshot: Snapshot?
+  ): State {
+    return if (simulatedPerfConfig.useInitializingState) Initializing else Selected(
+      NO_SELECTED_STANZA
+    )
+  }
+
+  @OptIn(WorkflowUiExperimentalApi::class)
+  override fun render(
+    renderProps: Poem,
+    renderState: State,
+    context: RenderContext
+  ): OverviewDetailScreen {
+    return when (renderState) {
+      Initializing -> {
+        context.runningWorker(Worker.from { Unit }, "initializing") {
+        action {
+          state = Selected(NO_SELECTED_STANZA)
+        }
+      }
+        isLoading.value = true
+        OverviewDetailScreen(overviewRendering = BackStackScreen(BlankScreen))
+      }
+      else -> {
+        val (stanzaIndex, currentStateIsLoading) = when (renderState) {
+          is ComplexCall -> renderState.payload to true
+          is Selected -> renderState.stanzaIndex to false
```

Comment:
```suggestion
          is ComplexCall -> Pair(renderState.payload,  true)
          is Selected -> Pair(renderState.stanzaIndex, false)
```

Pet peeve, I find `to` confusing outside of Map initialization.